secure/json-rpc/src/lib.rs

```diff
@@ -687,105 +687,178 @@ mod test {
         }
     }
 
-    // This function fuzzes a get_account_state_with_proof() response.
-    pub fn fuzz_account_state_response(data: &[u8]) {
-        if let Ok(account_state_response) = String::from_utf8(data.to_vec()) {
-            let ureq_response = Response::new(200, "Success", &account_state_response);
-            let _ = process_account_state_response(ureq_response);
+    // This generates an arbitrary response for the submit_transaction() JSON RPC API call.
+    prop_compose! {
+        fn arb_submit_transaction_response_impl(
+        )(
+            status in any::<u16>(),
+            status_text in any::<String>(),
+            id in any::<u64>(),
+            jsonrpc in any::<String>(),
+            result in arb_json_value(),
+        ) -> Response {
+            let response_body = SubmitTransactionResponse {
+                id,
+                jsonrpc,
+                result: Some(result),
+            };
+            let response_body =
+                serde_json::to_string::<SubmitTransactionResponse>(&response_body).unwrap();
+            Response::new(status, &status_text, &response_body)
         }
     }
 
-    // This function fuzzes a get_account_transaction() response.
-    pub fn fuzz_transaction_status_response(data: &[u8]) {
-        if let Ok(transaction_status_response) = String::from_utf8(data.to_vec()) {
-            let ureq_response = Response::new(200, "Success", &transaction_status_response);
-            let _ = process_transaction_status_response(ureq_response);
+    // This generates an arbitrary response for the get_account_transaction() JSON RPC API call.
+    prop_compose! {
+        fn arb_transaction_status_response_impl(
+        )(
+            status in any::<u16>(),
+            status_text in any::<String>(),
+            id in any::<u64>(),
+            jsonrpc in any::<String>(),
+            vm_status in arb_vm_status_view(),
+        ) -> Response {
+            let transaction_view = TransactionView {
+                vm_status
+            };
+            let response_body = TransactionViewResponse {
+                id,
+                jsonrpc,
+                result: Some(transaction_view),
+            };
+            let response_body =
+                serde_json::to_string::<TransactionViewResponse>(&response_body).unwrap();
+            Response::new(status, &status_text, &response_body)
         }
     }
+
+    pub fn arb_account_state_response() -> impl Strategy<Value = Response> {
+        arb_account_state_response_impl()
+    }
+
+    pub fn arb_submit_transaction_response() -> impl Strategy<Value = Response> {
+        arb_submit_transaction_response_impl()
+    }
+
+    pub fn arb_transaction_status_response() -> impl Strategy<Value = Response> {
+        arb_transaction_status_response_impl()
+    }
+
+    // This function generates an arbitrary serde_json::Value.
+    fn arb_json_value() -> impl Strategy<Value = Value> {
```

Comment:
so if I understand correctly you're trying to respond with garbage but valid json objects to submit?

Reply:
Yes exactly. The client logic doesn't actually ever use the response json result, but we may as well test what happens when one is returned :)